analyticskit/src/main/java/com/busybusy/analyticskit_android/AnalyticsEvent.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2016 - 2022 busybusy, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.busybusy.analyticskit_android
+
+import androidx.annotation.NonNull
+import java.io.Serializable
+
+/**
+ * Defines information that is needed to distribute the event to the registered analytics providers.
+ *
+ * @author John Hunt on 3/5/16.
+ */
+open class AnalyticsEvent(
+    @NonNull val name: String,
+    var attributes: MutableMap<String, Any>? = null,
+    var timed: Boolean = false,
+    var priorityLevel: Int = 0,
+) : Serializable {
+
+    /**
+     * Access the event name.
+     *
+     * @return the name of the custom event
+     */
+    fun name(): String {
+        return name
+    }
+
+    /**
+     * Adds an attribute to the event.
+     *
+     * @param attributeName the name of the attribute (should be unique)
+     * @param value         the [Object] to associate with the name given
+     * @return the [AnalyticsEvent] instance
+     */
+    fun putAttribute(attributeName: String, value: Any): AnalyticsEvent {
+        // guard clause - make sure the dictionary is initialized
+        if (attributes == null) {
+            attributes = LinkedHashMap()
+        }
+        attributes!![attributeName] = value
+        return this
+    }
+
+    /**
+     * Gets the priority of this event.
+     *
+     * @return the priority of the event. Returns `0` when [.setPriority] has not been called.
+     */
+    fun getPriority(): Int = priorityLevel
+
+    /**
+     * Sets the priority of the event. The event defaults to `0` when this method is not called.
+     *
+     *
+     * **Note:** It is up to the developer to define what priority scheme to use (if any).
+     *
+     * @param priorityLevel the priority the event should use
+     * @return the [AnalyticsEvent] instance (for builder-style convenience)
+     */
+    fun setPriority(priorityLevel: Int): AnalyticsEvent {
+        this.priorityLevel = priorityLevel
+        return this
+    }
+
```

Comment:
I had to remove the `getAttributes()` function - since the property is named `attributes` this would have created a JVM name clash for the generated getter method.

I chose to simply remove the `getAttributes()` function from the Kotlin version to avoid this problem.

Reply:
[`@JvmName`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-name/) instead maybe?

Reply:
Yes, thank you for pointing this out - `@JvmName` will solve this nicely.